lib/client.js

```diff
@@ -107,37 +103,38 @@ class Client extends ClientBase {
 
     const ret = new Duplex({
       autoDestroy: true,
-      writableHighWaterMark: 1,
+      highWaterMark: 1,
       read () {
         if (body) {
           body.resume()
         }
       },
       write (chunk, encoding, callback) {
-        if (req.push(chunk, encoding)) {
-          callback()
-        } else {
-          req[kResume] = callback
-        }
+        // TODO: The callback might not be invoked
+        // on error in pre Node 14.
+        // TODO: The callback might not correspond
+        // to a successful write to the socket.
+        req.write(chunk, encoding, callback)
       },
       final (callback) {
-        req.push(null)
-        callback()
+        // TODO: The callback might not be invoked
+        // on error in pre Node 14.
+        // TODO: The callback might not correspond
+        // to a successful flush of the socket.
```

Comment:
These notes probably don't have much if any impact in practical usage. But it might not strictly follow the expected API contract.

Reply:
Will require some further work (#208) to fully resolve but I don't think this is a blocker for release.